### PR TITLE
Make the Bind Method of Tensor more automatic

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

Function optimization
### PR changes

APIs
### Describe

Makes the Bind Method more automatic.

There these Change:

1. Replace the listed Tensor API by: ``linalg, math, logic, manipulation, search, stat, attribute`` modules which are related to Tensor operation, and the method of exhaustion is not suitable.
2. Fix some outdated documention of Tensor API, such as ``Tensor.detach``.
2. Remove ``paddle.to_variable`` from 2.0Rc.
3. Add unittest coverage of all Tensor's API, whose total number is about 150.

The relevant documentation is:
===

https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/tensor/creation/Tensor_cn.html#tensor
image

---
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/tensor_introduction.html
image

image

### CODE REVIEW

File: paddle/fluid/pybind/imperative.cc [hunk 1]
```diff
@@ -652,26 +652,22 @@ void BindImperative(py::module *m_ptr) {
...
             .. code-block:: python
 
-                import paddle.fluid as fluid
-                from paddle.fluid.dygraph.base import to_variable
-                from paddle.fluid.dygraph import Linear
+                import paddle
                 import numpy as np
+                paddle.disable_static()
 
                 data = np.random.uniform(-1, 1, [30, 10, 32]).astype('float32')
```
Comment (jzhang533) [inline paddle/fluid/pybind/imperative.cc]: 可以直接用`paddle.randn`，不需要用numpy创建再to_tensor

File: paddle/fluid/pybind/imperative.cc [hunk 2]
```diff
@@ -688,22 +684,22 @@ void BindImperative(py::module *m_ptr) {
         Examples:
              .. code-block:: python
 
-                import paddle.fluid as fluid
+                import paddle
                 import numpy as np
+                paddle.disable_static()
```
Comment (jzhang533) [inline paddle/fluid/pybind/imperative.cc:680]: 用2.0 API， 并且不需要numpy也能完成。

File: python/paddle/fluid/dygraph/math_op_patch.py [hunk 3]
```diff
@@ -85,15 +79,14 @@ def astype(self, dtype):
 
             .. code-block:: python
 
-                import paddle.fluid as fluid
+                import paddle
                 import numpy as np
 
                 x = np.ones([2, 2], np.float32)
```
Comment (jzhang533) [inline python/paddle/fluid/dygraph/math_op_patch.py]: 可以不需要numpy

File: python/paddle/fluid/dygraph/math_op_patch.py [hunk 4]
```diff
@@ -318,7 +273,12 @@ def __impl__(self):
             setattr(core.VarBase, method_name, method_impl)
     else:
         import paddle.tensor
-        for method_name in common_methods:
+        # Tensor method from module paddle.tensor
+        tensor_methods = paddle.tensor.linalg.__all__ + \
+            paddle.tensor.math.__all__ + paddle.tensor.logic.__all__ + \
```
Comment (jzhang533) [inline python/paddle/fluid/dygraph/math_op_patch.py]: 每行只写一个更好吧。

File: python/paddle/fluid/layers/math_op_patch.py [hunk 5]
```diff
@@ -372,7 +349,11 @@ def __impl__(self, other_var):
             setattr(Variable, method_name, method_impl)
     else:
         import paddle.tensor
-        for method_name in common_methods:
+        variabel_methods = paddle.tensor.linalg.__all__ + \
```
Comment (jzhang533) [inline python/paddle/fluid/layers/math_op_patch.py:352]: 确认一下，这里这样改，是否会引起paddle.fluid下的API的兼容性问题？

Comment (zhwesky2010) [reply python/paddle/fluid/layers/math_op_patch.py:352]: 这里只有paddle.fluid.Variable这个class的类方法发生了增加，原有的API不会发生改变

File: python/paddle/fluid/dygraph/math_op_patch.py (hunk 4 above)
Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/math_op_patch.py]: Done

File: paddle/fluid/pybind/imperative.cc (hunk 2 above)
Comment (zhwesky2010) [reply paddle/fluid/pybind/imperative.cc:680]: Done

File: python/paddle/fluid/dygraph/math_op_patch.py (hunk 3 above)
Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/math_op_patch.py]: Done

File: paddle/fluid/pybind/imperative.cc (hunk 1 above)
Comment (zhwesky2010) [reply paddle/fluid/pybind/imperative.cc]: Done

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by zhiqiu — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**